### ISSUE TITLE
Add macOS 13 runner image

### ISSUE THREAD

**Issue by Brett-Best** [actions/runner-images#6426]
### Tool name

macOS 13

### Tool license

https://www.apple.com/au/legal/sla/

### Add or update?

- [X] Add
- [ ] Update

### Desired version

13 RC (22A379)

### Brief description of tool

Latest major macOS version released by Apple.

### URL for tool's homepage

https://developer.apple.com/documentation/macos-release-notes/macos-13-release-notes

### Platforms where you need the tool

- [ ] Azure DevOps
- [X] GitHub Actions

### Runner images where you need the tool

- [ ] Ubuntu 18.04
- [ ] Ubuntu 20.04
- [ ] Ubuntu 22.04
- [ ] macOS 10.15
- [ ] macOS 11
- [ ] macOS 12
- [ ] Windows Server 2019
- [ ] Windows Server 2022

### Can this tool be installed during the build?

```bash
No
```

### Tool installation time in runtime

N/A

### Are you willing to submit a PR?

No

**Comment by EwoutH** [comment]
As for the timing of this issue, the [macOS 13 Release Candiate](https://developer.apple.com/news/releases/?id=10182022e) and [Xcode 14.1 Release Candiate](https://developer.apple.com/news/releases/?id=10182022h) were released today, along with an announcement that [Mac App Store submissions are now open for macOS Ventura](https://developer.apple.com/news/?id=vc82glbw).

This means developers will want to start testing on macOS 13, both locally and in CI.

**Comment by ddobranic** [comment]
@EwoutH , thank you for your request. New macOS-12 image with included Xcode 14.1 RC (14B47) is in release process currently.

**Comment by EwoutH** [comment]
This is the big one that needs to be done for macOS 13: https://github.com/actions/runner-images/pull/4062

**Comment by 3a4oT** [comment]
Do we have ETA on this? Would be nice to avoid last year's situation where we weren't able to upgrade to the latest version of Xcode because of latest macOS was not supported.

**Comment by mikhailkoliada** [comment]
I believe @Steve-Glass may provide more info on ETA

**Comment by erencelik** [comment]
Also need ETA, we need Ventura in order to proceed our CI&CD tests

**Comment by clayellis** [comment]
I don't want to just spam with "same" — but same. We have had to temporarily set up a self-hosted runner to unblock ourselves.

**Comment by halonsoluis** [comment]
With the official release of Ventura (macos13) and Xcode 14.1 this gets even more important. Could we have an ETA and a bit of a push? This ticket has been open for almost a month now.

**Comment by Brett-Best** [comment]
Two weeks later and no further information provided by GitHub ☹️

**Comment by dockay** [comment]
Bump here too, would be nice to have 10.13 asap.

**Comment by hassila** [comment]
Yes, this is a blocker for our projects using Swift 5.7 features - not possible to deploy those on GitHub runners (and we'd like to open source a few things but it'd be great with a working CI pipeline...). So +1.

**Comment by halonsoluis** [comment]
Hi @hassila, I had the same issue, but ended up using **macos-12** image

>       runs-on: macos-12

It has Xcode14, so it's OK to work with.

**Comment by hassila** [comment]
No, we use features which are only available with the Ventura Swift runtime (distributed actors, duration/instant/clock, ...) so macos-12 is a no-go unfortunately.

**Comment by Steve-Glass** [comment]
👋  We are targeting `Q2 2023 – Apr-Jun` for public beta. The public roadmap will be updated after December 15th with this objective. The plan is to open the beta runner image to all developers without any sign-up process.

**Comment by gereons** [comment]
@Steve-Glass thanks for the info. Are there any plans to speed this process up for the predictable upcoming releases of macOS?

**Comment by Brett-Best** [comment]
Wow that’s a long time away, while Xcode 14.3 beta isn’t out yet (perhaps next week) previous releases have required the next major OS version. This was an issue with both macOS 11 and macOS 12.

@gereons after delayed releases of macOS 11 and macOS 12, assurances were provided that it would be improved.
Based on the timeline provided, it appears it could actually be worse this time around with macOS 13. 😬 

I think it’s fair to conclude GitHub doesn’t prioritise macOS/iOS CI/CD at all.

The comments in https://github.com/actions/runner-images/issues/3649 are unfortunately still very relevant.

**Comment by halonsoluis** [comment]
Maybe time to consider other options from the competition, many customers of GitHub Actions that need this update will have no other choice.

**Comment by rudrankriyam** [comment]
If you prefer a CI/CD that prioritizes iOS/macOS development (macOS 13 Ventura and Xcode 14.2 already available!), give Codemagic a try.

https://codemagic.io/ios-continuous-integration

**Comment by clayellis** [comment]
For reference, the public roadmap issue that @Steve-Glass mentioned above is here: https://github.com/github/roadmap/issues/620

**Comment by billnbell** [comment]
+1 on getting this done! Ww needed it 3 months ago.

**Comment by LFMAKER** [comment]
Cool, github doesn't prioritize running github actions on macos, when they release new version of macOs, maybe upgrade to ventura. Definitely should get out of github actions.

This image should be available no later than 1 week after the release of ventura.

**Comment by Brett-Best** [comment]
> Wow that’s a long time away, while Xcode 14.3 beta isn’t out yet (perhaps next week) previous releases have required the next major OS version. This was an issue with both macOS 11 and macOS 12.

Xcode 14.3b1 is now out and it requires macOS 13.

**Comment by clayellis** [comment]
@Steve-Glass in an attempt to be productive and not just throw shade your team's way — can we maybe start some discourse around this issue? Some visibility into why it takes so long for GitHub to publish updated macOS runners might be helpful in quelling the mob. Is there some complex technical issue that we're all unaware of? Does it take a while to validate? Is it a planning or resources issue? Is there a blocking dependency on some other system? Is it virtual and/or physical hardware upgrades that need to happen first?

**Comment by BrianVTruong** [comment]
It'd be nice to have this sooner than later. I've been needing this for a while.

**Comment by BrentMifsud** [comment]
Is there any updated here? macOS 13 has been available for months now.

**Comment by billnbell** [comment]
I also think you all need to respond to us. What is the status?

**Comment by 3a4oT** [comment]
As said, Xcode 14.3 requires macOS 13, and we end up in the same trap as previous years. While I can understand (not really but acceptable) GitHub action CI, I don't get why Azure Pipelines (my company paying bills) are at the same level. It seems like they are using the same infrastructure, so why you can't do POC on  GitHub Actions first?

**Comment by martinjeret** [comment]
Disclaimer: I work at Codemagic

If people need to switch there are several providers including Codemagic and Bitrise that offer macOS Ventura images with Xcode 14.3. As of today Circle CI does not offer this AFAIK. Hope this helps people who need to build on Xcode 14.3 today.

**Comment by neoswallow** [comment]
I'm not sure about availability on CircleCI itself but they do have tart VM image with 14.3 beta. It was added soon after the beta was release.
I don't use their service, just tart on my machines.

**Comment by orschaef** [comment]
+1 We also really need a macOS 13 image. Our macOS app runs on 13 only ... what about updates on the current state?

**Comment by SPConrad** [comment]
> 👋 We are targeting `Q2 2023 – Apr-Jun` for public beta. The public roadmap will be updated after December 15th with this objective. The plan is to open the beta runner image to all developers without any sign-up process.

Can you help us understand why it takes 6 - 8 months for this to get to the Beta? stage? I'm not trying to be snarky, this is frustrating for a number of users and if you tried explaining some of this to us it could go a long way towards towards mitigating those frustrations.

**Comment by hassila** [comment]
Yeah, now it's a race whether macOS14 will ship as production before the macOS 13 runner makes it out of beta at GitHub - it's basically one year off in terms of schedule. To GitHub PM: Please start planning now for macOS 14 - it can't be a surprise when Apple announces it at WWDC this year. Then start testing with the beta developer versions and aim for shipping a few weeks after the production release in the fall. It is what your customers would expect you to do.

**Comment by diogot** [comment]
After two years of this we gave up, just bought a Mac mini and migrate to self hosted.
I don't know if GitHub (and Microsoft) see macOS as a hobby project or they have a NDA that forbid to discuss the issue openly.
Either way they should give their paid clients a clear position.

**Comment by Steve-Glass** [comment]
You are heard and we understand your frustration with the macOS runner image release cadence. 

>Can you help us understand why it takes 6 - 8 months for this to get to the Beta? stage? I'm not trying to be snarky, this is frustrating for a number of users and if you tried explaining some of this to us it could go a long way towards towards mitigating those frustrations.

@SPConrad

>Are there any plans to speed this process up for the predictable upcoming releases of macOS?

@gereons 

>After two years of this we gave up, just bought a Mac mini and migrate to self hosted.

**Comment by orschaef** [comment]
Funny - I saw that @Steve-Glass wrote something like "You are heard" without any updates (about one hour ago). Now the comment is gone. 😊

**Comment by Steve-Glass** [comment]
Thank you all  for reaching out and expressing your concerns. You are heard and we’ve read through all of your feedback. Additional context has been provided below to address frequent questions: 

**Why is macOS 13 not available yet?** 

- The delayed availability of macOS 13 is due to Apple’s [hardware compatibility requirement](https://support.apple.com/en-us/HT213264) for the image.
- Given the feedback provided during the macOS 12 release, our team was fully prepared to provide macOS 13 shortly after Apple’s announcement, but was not able to achieve this goal due to compatibility constraints.

**What does this mean for macOS 13 availability?** 

- Rest assured, GitHub has the required hardware for macOS 13. The public roadmap states [Q2 2023 - Apr-Jun](https://github.com/github/roadmap/issues/620) for the public beta. When the public beta is released, there won’t be a sign-up process like with macOS 12 and any developer can use it. 
- We understand the ambiguous timeline makes many developers uneasy. While I can’t provide a specific date, I can say that macOS 13 is a top priority and our team is actively working on the last set of requirements needed for launch. 

**How does this impact macOS 14 availability?** 

- Our team is not aware of any hardware compatibility requirements that would prevent us from prioritizing and providing the macOS 14 runner image in a timely manner. 

>In an attempt to be productive and not just throw shade your team's way — can we maybe start some discourse around this issue? Some visibility into why it takes so long for GitHub to publish updated macOS runners might be helpful in quelling the mob. Is there some complex technical issue that we're all unaware of? Does it take a while to validate? Is it a planning or resources issue? Is there a blocking dependency on some other system? Is it virtual and/or physical hardware upgrades that need to happen first?

@clayellis We appreciate the feedback. As stated above, the delay is almost entirely due to hardware compatibility constraints. 

>Can you help us understand why it takes 6 - 8 months for this to get to the Beta? stage? I'm not trying to be snarky, this is frustrating for a number of users and if you tried explaining some of this to us it could go a long way towards mitigating those frustrations.

@SPConrad It should not. I hope the response above addresses your concerns and demonstrates our current and future plans for macOS support. 

>To GitHub PM: Please start planning now for macOS 14 - it can't be a surprise when Apple announces it at WWDC this year. Then start testing with the beta developer versions and aim for shipping a few weeks after the production release in the fall. It is what your customers would expect you to do.

@hassila Thank you for your suggestion and for expressing your expectations regarding our planning for macOS 14. We are committed to improving our processes for providing Actions users with macOS runner images in a timely manner. This feedback has already taken this feedback into consideration for future planning. 

>thanks for the info. Are there any plans to speed this process up for the predictable upcoming releases of macOS?

@gereons Yes and it is a top priority.

> Funny - I saw that @Steve-Glass wrote something like "You are heard" without any updates (about one hour ago). Now the comment is gone. 😊

@orschaef I accidentally hit the comment button while drafting this message. I hope it helps provide additional context!

**Comment by hassila** [comment]
@Steve-Glass thanks for elaborating - it is easier to understand the delay if additional hardware is required. Hoping for Apple silicon runners coming up 😄 

Happy to hear your are listening and looking forward to see how it plays out now this fall with macOS 14 when the new hardware is in place.

**Comment by Steve-Glass** [comment]
> Thanks for the update. Do you expect the runners to be Apple Silicon based and/or Intel based?

@Brett-Best I expect macOS 14 to be available on Intel first. When [Apple silicon is available](https://github.com/github/roadmap/issues/528) this year, we intend to support both macOS 13 and macOS 14.

**Comment by svaldetero** [comment]
@Steve-Glass Thanks for the update. Do you know if there is a separate item to follow for AzureDevops Pipelines or will this item cover both?

**Comment by orschaef** [comment]
@Steve-Glass  Thanks for the detailed explanation. It helps to understand the delay. Since my comment about your unfinished message does not add any value here, I deleted it.

**Comment by wAuner** [comment]
I hope that macOS 13 will be available sooner rather than later, considering the new Xcode version 14.3, already available as RC, will only support macOS 13+. This puts us in a tough spot because we either can't update Xcode or risk working with different compiler versions. Either way, our adoption of new C++ features is blocked by our Azure CI. That's nothing we anticipated.

**Comment by erikng** [comment]
If it's a matter of hardware, why can't you offer a few machines running macOS 13 and just have an expectation that the runners may be in a queue for a long time? I'd much rather have slow access to macOS 13 and enable Xcode 14.3 than be stuck with macOS 12 and Xcode 14.2.

There are some great features in Swift 5.8 that I would love to use, but GitHub is effectively blocking me from utilizing them. It's great that there aren't concerns for macOS 14, but by then we will likely be on Swift 5.9 or 6.0.

**Comment by mariusfanu-metrobank** [comment]
Any updates on this? 

We're looking forward to macOS 13 images so that we can use the Xcode 14.3, and Swift 5.8

**Comment by orschaef** [comment]
@mariusfanu-metrobank I think it is not that helpful to ping every week now. We got a detailed message from @Steve-Glass - which is one week old: 

https://github.com/actions/runner-images/issues/6426#issuecomment-1479974889

I feel the same pain as many of us here. But we also need some patience I guess.

**Comment by Brett-Best** [comment]
I updated the top of the issue to include a bigger sized update message... hopefully people don’t miss seeing it now 🤞🏻

**Comment by scannillo** [comment]
👋 Hello - is there a timeline for when MacOS 13 & Xcode 14.3 will be available?

**Comment by lodizn** [comment]
In a weird position here!! Just finished a contract and acquired 3 Mac minis (MacOS13 )  with MacStadium and realized we don't have a runner image ;( never thought this would be an issue!! is there any workaround for time being??

**Comment by landsman** [comment]
We are now facing issue with missing Xcode 14.3.
Can you please provide image with this version? 🙏 

> That's because `isInspectable` is available since iOS 16.4 / Swift 5.8 which is shipped with Xcode 14.3 and GitHub Actions are still on Xcode 14.2

**Comment by slajar** [comment]
Please add a mac-13 image for azure. It is getting urgent.

**Comment by ssbnirma** [comment]
@Steve-Glass ,  Please share some updates on when this will be available for customers? 

Currently my customer is blocked on this for several applications and are trying to work around it the best they can, however the latest version of Xcode requires macOS 13/Ventura which isn't yet available as a runner image despite being released publicly in Oct 2022 and in beta June 2022. .

**Comment by BenjaminBhuyanYoung** [comment]
> > why can't you offer a few machines running macOS 13 and just have an expectation that the runners may be in a queue for a long time? I'd much rather have slow access to macOS 13 and enable Xcode 14.3 than be stuck with macOS 12 and Xcode 14.2.
> 
> @eriking This is wonderful feedback! Will take this into consideration as we finalize preparations. If runner availability is a concern, our team will set queue time expectations up front.

Yes please, this would help us a lot.

**Comment by gghart** [comment]
In the MacOS-13 image, can you make sure to include msbuild 17?

Even though we don't appear to have any blocking dependencies on new libraries in MacOS 13, we can't compile .Net 7 code on MacOS-12 because it only has msbuild 16 which can't target .Net 7.

Like many people commenting above - this seems ridiculously late and is blocking many people..

**Comment by orschaef** [comment]
Would it perhaps be possible to be a little more precise about the time period? We are now already between Apr - June and it would be helpful for many at least to know whether they now still have to wait 1 week or 2 months.

**Comment by Steve-Glass** [comment]
https://github.blog/changelog/2023-04-24-github-actions-macos-13-is-now-available/

**Comment by Steve-Glass** [comment]
macOS 13 is also available for Azure DevOps. Communications will be out this afternoon (EST).

**Comment by 3a4oT** [comment]
Hey @Steve-Glass, thanks for the update. FYI. Seems like the link with "software list" [doesn't work](https://github.com/actions/runner-images/blob/main/images/macos/macos-13-Readme.md).

**Comment by billnbell** [comment]
OK the Azure Node NVM does not work in the new MacOS 13.

$(Agent.HomeDirectory)/scripts/select-node-version.sh node18 

It does not find the `select-node-version.sh` script. Unsure why.

**Comment by Brett-Best** [comment]
@billnbell raise a new issue

**Comment by billnbell** [comment]
OK but nvm was there in the past.

**Comment by remyjette** [comment]
I believe this has the same root cause as https://github.com/actions/setup-python/issues/658 . That got sent over to setup-python because I was trying Python, but I don't think that's the right place for it... The issue is the hostedtoolcache isn't populated at all.

**Comment by gghart** [comment]
Are the msbuild and nuget tasks expected to work on a MacOS-13 agent? 'dotnet publish' runs fine and can build our native Mac Cocoa app, but msbuild and nuget tasks die immediately with a mono not found error.

MSBuild fails with:
##[error]MSBuild or xbuild (Mono) were not found on the macOS or Linux agent.

NuGet fails with:
##[error]Error: Unable to locate executable file: 'mono'. Please verify either the file path exists or the file can be found within a directory specified by the PATH environment variable. Also check the file mode to verify the file is executable.


Not really clear if these tasks are supposed to be used anymore with .Net 5.0+?  We used them successfully with .Net 4.7.2, but they've never worked on .Net 6.0 or .Net 7.0. The documentation doesn't seem to say you shouldn't use them anymore, but they don't work on MacOS-12 or MacOS-13 images - but the error is different on both images. On MacOS-12, the errors are related to the TargetFramework attribute, on MacOS-13, they fail with the mono not found error.

**Comment by billnbell** [comment]
Agreed we need toolcache 

```
I believe the issue is that in the macOS-13 image, the hostedtoolcache has not been populated. For example, https://github.com/actions/runner-images/blob/main/images/macos/toolsets/toolset-13.json has no toolcache entry, compared to https://github.com/actions/runner-images/blob/main/images/macos/toolsets/toolset-12.json which does.```

**Comment by Steve-Glass** [comment]
> Hey @Steve-Glass, thanks for the update. FYI. Seems like the link with "software list" [doesn't work](https://github.com/actions/runner-images/blob/main/images/macos/macos-13-Readme.md).

The page should be available now. Please submit a new issue for any concerns with existing software, or requests for new software.

**Comment by mikhailkoliada** [comment]
Hi All! 

We are aware that macOS-13 is still missing lots of software, we are also aware of what is considered the most important and desired ones. We are working on re-adding the software back! Please remember that it is still in the beta phase and things will change.

Thanks for your patience! :)
